pkg/gitconfig/configs.go

```diff
@@ -0,0 +1,358 @@
+			cs.worktree = c
+		}
+	}
+	cs.worktree.noWrites = cs.NoWrites
+
+	// load any env vars
+	cs.env = LoadConfigFromEnv(cs.EnvPrefix)
+
+	return cs
+}
+
+func globalConfigFile() string {
+	// $XDG_CONFIG_HOME/git/config
+	return filepath.Join(appdir.UserConfig(), "config")
+}
+
+// loadGlobalConfigs will try to load the per-user (Git calls them "global") configs.
+// Since we might need to try different locations but only want to use the first one
+// it's easier to handle this in it's own method.
+func (c *Configs) loadGlobalConfigs() string {
+	for _, p := range []string{
+		globalConfigFile(),
+		// ~/.gitconfig
+		c.GlobalConfig,
+	} {
+		// GlobalConfig might be set to an empty string to disable it
+		// and instead of the XDG_CONFIG_HOME path only.
+		if p == "" {
+			continue
+		}
+		cfg, err := LoadConfig(p)
+		if err != nil {
+			debug.Log("failed to load global config from %s", p)
+
+			continue
+		}
+
+		debug.Log("loaded global config from %s", p)
+		c.global = cfg
+
+		return p
+	}
+
+	debug.Log("no global config found")
+
+	// set the path in case we want to write to it (create it) later
+	c.global = &Config{
+		path: globalConfigFile(),
+	}
+
+	return ""
+}
+
+// HasGlobalConfig indicates if a per-user config can be found.
+func HasGlobalConfig() bool {
+	c := &Configs{}
+
+	return c.loadGlobalConfigs() != ""
+}
+
+// Get returns the value for the given key from the first location that is found.
+// Lookup order: env, worktree, local, global, system and presets.
+func (c *Configs) Get(key string) string {
+	for _, cfg := range []*Config{
+		c.env,
+		c.worktree,
+		c.local,
+		c.global,
+		c.system,
+		c.Preset,
+	} {
+		if cfg == nil || cfg.vars == nil {
+			continue
+		}
+		if v, found := cfg.vars[key]; found {
+			return v
+		}
+	}
+
+	debug.Log("no value for %s found", key)
+
+	return ""
+}
+
+// GetGlobal specifically ask the per-user (global) config for a key.
+func (c *Configs) GetGlobal(key string) string {
+	if c.global == nil {
+		return ""
+	}
+
+	if v, found := c.global.vars[key]; found {
+		return v
+	}
+
+	debug.Log("no value for %s found", key)
+
+	return ""
+}
+
+// GetLocal specifically asks the per-directory (local) config for a key.
+func (c *Configs) GetLocal(key string) string {
+	if c.local == nil {
+		return ""
+	}
+
+	if v, found := c.local.vars[key]; found {
+		return v
+	}
+
+	debug.Log("no value for %s found", key)
+
+	return ""
+}
+
+// IsSet returns true if this key is set in any of our configs.
+func (c *Configs) IsSet(key string) bool {
+	for _, cfg := range []*Config{
+		c.env,
+		c.worktree,
+		c.local,
+		c.global,
+		c.system,
+		c.Preset,
+	} {
+		if cfg.IsSet(key) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// SetLocal sets (or adds) a key only in the per-directory (local) config.
+func (c *Configs) SetLocal(key, value string) error {
+	if c.local == nil {
+		if c.workdir == "" {
+			return fmt.Errorf("no workdir set")
+		}
+		c.local = &Config{}
```

Comment:
Shouldn't this set the path of the local config?

Reply:
Absolutely. Thank you!